src/server/routes/users.js

```diff
@@ -23,87 +22,116 @@ router.post('/register', (req, res) => {
 			errors : errors
 		});
 	} else {
-    const firstName = req.body.firstName;
-		const lastName 	= req.body.lastName;
+    const first_name = req.body.firstName;
+		const last_name 	= req.body.lastName;
 		const username 	= req.body.username;
 		const email 		= req.body.email;
 		const password 	= req.body.password;
     const accounts 	= req.body.accounts;
-    
-    User.checkIfAvailable('username', username)
-    .then(() => {
-      return User.checkIfAvailable('email', email);
+    let passHash = '';
+
+    bcrypt.genSalt(10, (err, salt) => {
+			if(err) throw(error);
+
+			bcrypt.hash(password, salt, (error, hash) => {
+				if(error) throw(error);
+				
+				passHash = hash;
+			});
+		});
+
+    queries.findUsername(username)
+    .then(results => {
+      if(results.length > 0) {
+        throw('Username already exists');
+      } else {
+        return queries.findEmail(email);
+      }
     })
-    .then(() => {
-      return Counter.getNextSequenceValue();
+    .then(results => {
+      if(results.length > 0) {
+        throw('Email already exists');
+      } else {
+        const newUser = {
+          first_name: first_name
+        , last_name: last_name
+        , username: username
+        , email: email
+        , pass_hash: passHash
+        };
+        return queries.createUser(newUser);
+      }
     })
-    .then(nextSequenceValue => {
-      const newUser = new User({
-        _id: nextSequenceValue
-      , firstName: firstName
-      , lastName: lastName
-      , username: username
-      , email: email
-      , password: password
-      });
-
-      return User.createUser(newUser);
+    .then(() => {
+      return queries.getUserIdBy('username', username);
     })
-    .then(newUserObject => {
+    .then(result => {
+      const user_id = result[0].user_id;
```

Comment:
what if no user was found?

Reply:
This is called just after creating the very user we're searching for. Since the db auto-increments the id, we do not know what the id value will be before the insertion, so that is the reason searching for the new user is necessary. If the previous promises are do not reject, this should never happen.